Initial conversion of main augur package to uv

JohnStrunk opened the pull request:
**Description**
- This revives #3148, targeting main.
- This supersedes (the uv portion of) #3198

This PR fixes #

**Notes for Reviewers**

**Signed commits**
- [ ] Yes, I signed my commits.

<!--
Thank you for contributing to CHAOSS projects! 

Contributing Conventions:
1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's [contribution conventions](https://github.com/chaoss/augur/blob/main/CONTRIBUTING.md) upfront, the review process will be accelerated and your PR merged more quickly.
-->

sgoggins commented:
@JohnStrunk / @Ulincsys : 

@Ulincsys - Ultimately, when I went with the entry being blank, it let me through and created the `.get-credentials` file at Augur's root. I wonder if that is the expectation and something just went awry?

`uv` does seem to behave differently on Ubuntu and OSX using bare metal. On OSX I can run install/rebuild commands like this: 
- OSX: `make install` or `make rebuild`
- Ubuntu: `uv run make install` or `uv run make rebuild` 

Its probably an idiosyncracy of `uv` and different OS's ... returning back to my earlier pondering of support for both bare metal and docker ... 

I also got something of an odd and SIMILAR (identical?) result on bare metal Ubuntu: 

```bash
The Facade data collection worker will clone repositories to this machine to run its analysis.
Please select a new or existing directory for the Facade worker to use:

Facade worker directory: /home/sean/uv-test-repos

File .git-credentials does not exist. Creating it...
touch: cannot touch '/home/sean/uv-test-repos/.git-credentials': No such file or directory
User sean does not have permission to write to /home/sean/uv-test-repos/.git-credentials
Please select another location
Facade worker directory: /home/sean/uv-test-repos

File .git-credentials does not exist. Creating it...
touch: cannot touch '/home/sean/uv-test-repos/.git-credentials': No such file or directory
User sean does not have permission to write to /home/sean/uv-test-repos/.git-credentials
Please select another location
Facade worker directory: 
You entered a blank line, are you sure?
enter 'yes' to continue, or enter the intended value: yes
realpath: missing operand
Try 'realpath --help' for more information.

File .git-credentials does not exist. Creating it...
touch: cannot touch '/.git-credentials': Permission denied
User sean does not have permission to write to /.git-credentials
Please select another location
Facade worker directory: 
You entered a blank line, are you sure?
enter 'yes' to continue, or enter the intended value: yes
realpath: missing operand
Try 'realpath --help' for more information.

File .git-credentials does not exist. Creating it...
touch: cannot touch '/.git-credentials': Permission denied
User sean does not have permission to write to /.git-credentials
Please select another location
Facade worker directory: 
You entered a blank line, are you sure?
enter 'yes' to continue, or enter the intended value: 
You entered a blank line, are you sure?
enter 'yes' to continue, or enter the intended value: yes
realpath: missing operand
Try 'realpath --help' for more information.

File .git-credentials does not exist. Creating it...
touch: cannot touch '/.git-credentials': Permission denied
User sean does not have permission to write to /.git-credentials
Please select another location
Facade worker directory: 
You entered a blank line, are you sure?
enter 'yes' to continue, or enter the intended value: 
You entered a blank line, are you sure?
enter 'yes' to continue, or enter the intended value: eys

File .git-credentials does not exist. Creating it...
touch: cannot touch '/home/sean/github/augur-uv-test/eys/.git-credentials': No such file or directory
User sean does not have permission to write to /home/sean/github/augur-uv-test/eys/.git-credentials
Please select another location
Facade worker directory: yes

File .git-credentials does not exist. Creating it...
touch: cannot touch '/home/sean/github/augur-uv-test/yes/.git-credentials': No such file or directory
User sean does not have permission to write to /home/sean/github/augur-uv-test/yes/.git-credentials
Please select another location
Facade worker directory: /home/sean/dog/

File .git-credentials does not exist. Creating it...
touch: cannot touch '/home/sean/dog/.git-credentials': No such file or directory
User sean does not have permission to write to /home/sean/dog/.git-credentials
Please select another location
Facade worker directory: /home/sean

File .git-credentials does not exist. Creating it...
Permission check passed for /home/sean/.git-credentials
That directory already exists. Use it? [Y/n]: Y
```

JohnStrunk commented:
- I've added a test for macOS that at least installs the dependencies. This catches the xgboost problem
- I've also upgraded xgboost, and added a constraint for graphql-server-core that was breaking the e2e test.

sgoggins commented:
> * I've added a test for macOS that at least installs the dependencies. This catches the xgboost problem
> 
>     * I've also upgraded xgboost, and added a constraint for graphql-server-core that was breaking the e2e test.

Thank you, @JohnStrunk !! We appreciate you!!

sgoggins reviewed (approved):
LGTM

Ulincsys commented on .python-version:
Does this force the project to run on 3.11? Is this a strict requirement, or a preference?

JohnStrunk replied:
(expanding on my answer from slack)
The `.python-version` file [determines the version of python](https://docs.astral.sh/uv/guides/projects/#python-version) that will be used by uv when building the venv for the project. It's also used in the CI workflow as input to the `setup-python` actions to determine the version of Python used there.

It's just providing the "default version", not mandating a minimum. The minimum is set in `pyproject.toml`. That happens to be 3.10 due to the xgboost version upgrade (68431f1e3adb22cc756d1e7bf014b355edb41fd5). It was 3.9 prior to that.

When I created the `.python-version` file, I went w/ the version we're using for the docker containers since that was known to work.
